### PR TITLE
Feature/cards : Improving overall UI design

### PR DESCRIPTION
**Pull request by JanStaszewiczEPFL**
# Description

This PR includes the following updates and improvements:

1. **Removed the top bar from a screen**:
   - Simplified the layout by removing the top bar for a cleaner UI.

2. **Updated profile card design**:
   - Added a new tag display design for profile cards.
   - Updated card colors to improve visual appeal.
   - Changed text styles for better readability and consistency.

3. **Added tests for the new tag display**:
   - Implemented unit tests to ensure correct rendering of the new tag display.
   - Verified proper integration of the tags with profile cards.
   - Covered edge cases to improve overall test coverage.

4. **Tweaked colors of the light theme**:
   - Adjusted primary and secondary color shades for enhanced contrast.
   - Updated background and text colors for improved readability and aesthetics.

# Screenshots

### CODE REVIEW

File: app/src/main/java/com/github/se/icebreakrr/ui/sections/Settings.kt [hunk 1]
```diff
@@ -101,8 +105,8 @@ fun SettingsScreen(
   val myProfile = profilesViewModel.selfProfile.collectAsState()
 
   Scaffold(
-      topBar = { TopBar("Settings") },
       modifier = Modifier.testTag("settingsScreen").fillMaxSize(),
+      topBar = { TopBar("Settings") },
```
Comment (BotondAKovacs) [inline app/src/main/java/com/github/se/icebreakrr/ui/sections/Settings.kt:109]: Nitpick: hardcoded string could be moved to strings.xml

File: app/src/main/java/com/github/se/icebreakrr/ui/sections/shared/ProfileCard.kt [hunk 2]
```diff
@@ -26,25 +27,77 @@ import androidx.compose.ui.unit.sp
...
+      // If there are more than 3 tags, add "..." at the end
+      if (tags.size > TAKE_TAGS) {
+        Text(
+            text = "...",
+            fontSize = TAGS_FONT_SIZE,
+            fontWeight = FontWeight.SemiBold,
+            color = MaterialTheme.colorScheme.primaryContainer,
+            modifier =
+                Modifier.padding(2.dp)
+                    .clip(RoundedCornerShape(25.dp)) // Rounded corners for the ellipsis
```
Comment (BotondAKovacs) [inline app/src/main/java/com/github/se/icebreakrr/ui/sections/shared/ProfileCard.kt:87]: Nitpick: magic number

File: app/src/main/java/com/github/se/icebreakrr/ui/sections/shared/ProfileCard.kt [hunk 3]
```diff
@@ -26,25 +27,77 @@ import androidx.compose.ui.unit.sp
...
+                    .background(
+                        MaterialTheme.colorScheme.secondaryContainer) // Tag background color
+                    .padding(horizontal = 10.dp, vertical = 4.dp) // Padding inside the ellipsis
```
Comment (BotondAKovacs) [inline app/src/main/java/com/github/se/icebreakrr/ui/sections/shared/ProfileCard.kt:90]: Nitpick: magic number

File: app/src/main/java/com/github/se/icebreakrr/ui/sections/shared/ProfileCard.kt [hunk 4]
```diff
@@ -26,25 +27,77 @@ import androidx.compose.ui.unit.sp
...
+            )
+      }
+    }
+  } else {
+    Text(
+        text = "Tap to preview profile",
```
Comment (BotondAKovacs) [inline app/src/main/java/com/github/se/icebreakrr/ui/sections/shared/ProfileCard.kt:96]: Nitpitck: hardcoded string

File: app/src/main/java/com/github/se/icebreakrr/ui/sections/Settings.kt [hunk 5]
```diff
@@ -222,13 +228,17 @@ fun ToggleOptionBox(
...
+              containerColor = MaterialTheme.colorScheme.background,
+              contentColor = MaterialTheme.colorScheme.onSecondary)) {
         Row(
             modifier = Modifier.fillMaxWidth().padding(CARD_PADDING),
             verticalAlignment = Alignment.CenterVertically) {
-              Text(label)
+              Icon(
+                  imageVector = Icons.Filled.Place,
+                  contentDescription = "",
+                  modifier = Modifier.padding(end = 8.dp) // Add padding to the right of the icon
```
Comment (BotondAKovacs) [inline app/src/main/java/com/github/se/icebreakrr/ui/sections/Settings.kt:239]: Nitpick: magic number

File: app/src/main/java/com/github/se/icebreakrr/ui/sections/Settings.kt (hunk 1 above)
Comment (JanStaszewiczEPFL) [reply app/src/main/java/com/github/se/icebreakrr/ui/sections/Settings.kt:109]: Yes I'll do it

File: app/src/main/java/com/github/se/icebreakrr/ui/sections/shared/ProfileCard.kt [hunk 6]
```diff
@@ -26,25 +27,77 @@ import androidx.compose.ui.unit.sp
...
+    FlowRow(modifier = Modifier.fillMaxWidth().padding(TEXT_SPACER_PADDING)) {
+      // Create a Text for each tag, limiting to the calculated number of tags
+      tags.take(tagsToDisplay).forEachIndexed { index, tag ->
+        Text(
+            text = "#$tag",
+            fontSize = TAGS_FONT_SIZE,
+            fontWeight = FontWeight.SemiBold,
+            color = MaterialTheme.colorScheme.primaryContainer,
+            modifier =
+                Modifier.padding(2.dp)
```
[the remark this answers is not in the document]
Comment (JanStaszewiczEPFL) [reply app/src/main/java/com/github/se/icebreakrr/ui/sections/shared/ProfileCard.kt:71]: I need to exterminate all magic numbers 🤖

File: app/src/main/java/com/github/se/icebreakrr/ui/sections/shared/ProfileCard.kt [hunk 7]
```diff
@@ -26,25 +27,77 @@ import androidx.compose.ui.unit.sp
...
+                    .clip(RoundedCornerShape(25.dp)) // Rounded corners for each tag
+                    .background(
+                        MaterialTheme.colorScheme.secondaryContainer) // Tag background color
+                    .padding(horizontal = 10.dp, vertical = 4.dp) // Padding inside the tag
```
[the remark this answers is not in the document]
Comment (JanStaszewiczEPFL) [reply app/src/main/java/com/github/se/icebreakrr/ui/sections/shared/ProfileCard.kt:75]: Magic numbers make code inefficient 🤖 we need to exterminate them as soon as possible 🤖🤖

File: app/src/main/java/com/github/se/icebreakrr/ui/sections/shared/ProfileCard.kt [hunk 8]
```diff
@@ -26,25 +27,77 @@ import androidx.compose.ui.unit.sp
...
+
+      // If there are more than 3 tags, add "..." at the end
+      if (tags.size > TAKE_TAGS) {
+        Text(
+            text = "...",
+            fontSize = TAGS_FONT_SIZE,
+            fontWeight = FontWeight.SemiBold,
+            color = MaterialTheme.colorScheme.primaryContainer,
+            modifier =
+                Modifier.padding(2.dp)
```
[the remark this answers is not in the document]
Comment (JanStaszewiczEPFL) [reply app/src/main/java/com/github/se/icebreakrr/ui/sections/shared/ProfileCard.kt:87]: This is intolerable, my system has missed magic numbers again 🤖🤖

File: app/src/main/java/com/github/se/icebreakrr/ui/sections/shared/ProfileCard.kt (hunk 2 above)
Comment (JanStaszewiczEPFL) [reply app/src/main/java/com/github/se/icebreakrr/ui/sections/shared/ProfileCard.kt:88]: This is too much ! My system need a reboot to focus on those MaGiCC Nu..Nu..mbers 🤖⚡️

File: app/src/main/java/com/github/se/icebreakrr/ui/sections/shared/ProfileCard.kt (hunk 3 above)
Comment (JanStaszewiczEPFL) [reply app/src/main/java/com/github/se/icebreakrr/ui/sections/shared/ProfileCard.kt:91]: AGAIN!!!! 🤖

File: app/src/main/java/com/github/se/icebreakrr/ui/sections/shared/ProfileCard.kt (hunk 4 above)
Comment (JanStaszewiczEPFL) [reply app/src/main/java/com/github/se/icebreakrr/ui/sections/shared/ProfileCard.kt:97]: I'll place this in xml values!

File: app/src/main/java/com/github/se/icebreakrr/ui/sections/Settings.kt (hunk 5 above)
Comment (JanStaszewiczEPFL) [reply app/src/main/java/com/github/se/icebreakrr/ui/sections/Settings.kt:243]: I hope it's the last one, sorry for the malfunction 🤖

**Review by BotondAKovacs — COMMENTED** [review]
This is my last nitpick, then you're good. Thanks!

**Review by BotondAKovacs — APPROVED** [review]
LGTM!

**Review by BotondAKovacs — APPROVED** [review]
